### PR TITLE
[G-API] Validate RMat adapter in executor

### PR DESCRIPTION
**Pull request by TolyaTalamanov**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [ ] I agree to contribute to the project under Apache 2 License.
- [ ] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [ ] The PR is proposed to proper branch
- [ ] There is reference to original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### Build configuration
```
force_builders=Custom,Custom Win,Custom Mac
build_gapi_standalone:Linux x64=ade-0.1.1f
build_gapi_standalone:Win64=ade-0.1.1f
build_gapi_standalone:Mac=ade-0.1.1f
build_gapi_standalone:Linux x64 Debug=ade-0.1.1f

Xbuild_image:Custom=centos:7
Xbuildworker:Custom=linux-1
build_gapi_standalone:Custom=ade-0.1.1f

build_image:Custom=ubuntu-openvino-2021.3.0:20.04
build_image:Custom Win=openvino-2021.2.0
build_image:Custom Mac=openvino-2021.2.0

test_modules:Custom=gapi,python2,python3,java
test_modules:Custom Win=gapi,python2,python3,java
test_modules:Custom Mac=gapi,python2,python3,java

buildworker:Custom=linux-1
# disabled due high memory usage: test_opencl:Custom=ON
test_opencl:Custom=OFF
test_bigdata:Custom=1
test_filter:Custom=*
```

### CODE REVIEW

**Comment by TolyaTalamanov** [comment]
@rgarnov Could you have a look ?

File: modules/gapi/src/executor/gexecutor.cpp [hunk 1]
```diff
@@ -159,8 +159,8 @@ void writeBackExec(const Mag& mag, const RcDesc &rc, GRunArgP &g_arg)
         // a real copy (add a pass to StreamingBackend?)
         auto& out_mat = *util::get<cv::Mat*>(g_arg);
         const auto& rmat = mag.template slot<cv::RMat>().at(rc.id);
-        auto mag_data = rmat.get<RMatAdapter>()->data();
-        if (out_mat.data != mag_data) {
+        auto* adapter = rmat.get<RMatAdapter>();
+        if (adapter != nullptr && out_mat.data != adapter->data()) {
```
Comment (alalek) [inline modules/gapi/src/executor/gexecutor.cpp:163]: > `adapter != nullptr`

Should this to be an assertion with raised exception in case of null?

Comment (TolyaTalamanov) [reply modules/gapi/src/executor/gexecutor.cpp:163]: No
`adapter == nullptr` is a normal case.
We need to do copy only if adapter is RMatAdapter and `out_mat.data != adapter->data()`

**Comment by TolyaTalamanov** [comment]
@rgarnov Could you have a look, please ?

**Review by rgarnov — APPROVED** [review]

**Pull request merged**